<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>441</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Morse Decoder</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Detected morse code:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="wpmLabel">
       <property name="text">
        <string>Detected WPM:</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="morseText"/>
   </item>
   <item>
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Decoded text:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPlainTextEdit" name="decodedText"/>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonKey">
       <property name="text">
        <string>Key</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="baseSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Text to play back as morse code:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTextEdit" name="textForPlayback"/>
   </item>
   <item>
    <widget class="QTextEdit" name="codedText"/>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonPlay">
       <property name="text">
        <string>Play back</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>onListen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>377</x>
     <y>285</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonKey</sender>
   <signal>pressed()</signal>
   <receiver>Dialog</receiver>
   <slot>onKeydown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>32</x>
     <y>288</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButtonKey</sender>
   <signal>released()</signal>
   <receiver>Dialog</receiver>
   <slot>onKeyup()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>274</y>
    </hint>
    <hint type="destinationlabel">
     <x>49</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onListen()</slot>
  <slot>onKeydown()</slot>
  <slot>onKeyup()</slot>
 </slots>
</ui>
